src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Circle extends Figure {
+    private int radius;
+
+    public Circle() {
+    }
+
+    public Circle(Random random) {
+        this.radius = random.nextInt(100);
+        setColor(ColorSupplier.getRandomColor());
+    }
```

Comment:
Figure implementations constructors should have color and their parameters in the constructor, nothing else
There also should be no logic or calls to other classes, just the initialization of the object
```suggestion
    public Circle(int radius, Color color) {
        this.radius = radius;
        this.color = color;
    }
```